src/components/shared/DetailsWidget/SafeTransactionDetail/components/Value.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import Numeral from '~shared/Numeral';
+import Avatar from '~shared/Avatar';
+import Icon from '~shared/Icon';
+import { Token, SafeTransaction } from '~types';
+import { TokenType } from '~gql';
+
+import { MSG } from '../../detailsWidgetConfig';
+
+import widgetStyles from '../../DetailsWidget.css';
+import styles from '../SafeTransactionDetail.css';
+
+const renderTokenIcon = (tokenData: Token) => {
+  if (tokenData.name === 'Ether') {
+    return (
+      <Icon className={styles.ether} name="ether-purple" title="Ether Logo" />
+    );
+  }
+
+  if (tokenData.type === TokenType.Erc20) {
+    return (
+      <Avatar
+        className={styles.tokenAvatar}
+        /* avatarURL={''} */
```

Comment:
this was commented out as the `Avatar` component no longer exposes a property called `avatarURL` I have changed it so that the `avatar` property itself is set to the token thumbnail... is this the correct behaviour? https://github.com/JoinColony/colonyCDapp/pull/900/commits/b46306d5db4a3c034dbeef94c845d983a7a986af

Reply:
Yep, looks good to me! Even better I've just noticed that we can reuse the `TokenIcon` component as it already takes care of how the data is passed to `Avatar` when it's token data.

Reply:
oh nice one thats a nice refactor thanks for the pointer